dynamic_update.php

```diff
@@ -175,25 +176,18 @@ function valid_ip_address($ip)
     return status_exit('notfqdn');
 }
 
-$user_query = "
-    SELECT
-        users.id
-    FROM
-        users, perm_templ, perm_templ_items, perm_items
-    WHERE
-        users.username = '$username'
-        AND users.password = '$password'
-        AND users.active = 1
-        AND perm_templ.id = users.perm_templ
-        AND perm_templ_items.templ_id = perm_templ.id
-        AND perm_items.id = perm_templ_items.perm_id
-        AND (
-                perm_items.name = 'zone_content_edit_own'
-                OR perm_items.name = 'zone_content_edit_others'
-        )
-";
-$user = $db->queryRow($user_query);
-if (!$user) {
+$user = $db->queryRow("SELECT users.id, users.password FROM users, perm_templ, perm_templ_items, perm_items WHERE users.username=" . $db->quote($auth_username, 'text') . " AND users.active=1 AND perm_templ.id = users.perm_templ AND perm_templ_items.templ_id = perm_templ.id AND perm_items.id = perm_templ_items.perm_id AND (perm_items.name = 'zone_content_edit_own' OR perm_items.name = 'zone_content_edit_others')");
+
+if ($user) {
+    if (Poweradmin\Password::verify($auth_password, $user['password'])) {
+        if (Poweradmin\Password::needs_rehash($rowObj['password'])) {
```

Comment:
maybe $user should be here instead of $rowObj?

Reply:
thanks